src/discof/replay/fd_replay_tile.c

```diff
@@ -441,6 +442,73 @@ publish_stake_weights( fd_replay_tile_ctx_t * ctx,
   }
 }
 
+static void
+replay_block_finalize( fd_replay_tile_ctx_t * ctx,
+                       fd_stem_context_t *    stem,
+                       fd_runtime_block_info_t * runtime_block_info ) {
+
+  /* TODO: Currently this is being done out of the exec tile. This
+     should eventually be moved to the privleged writer tiles which have
+     write access into the runtime spad. */
+
+    fd_accounts_hash_task_data_t * task_data = NULL;
+    fd_runtime_block_execute_finalize_start( ctx->slot_ctx, ctx->runtime_spad, &task_data, ctx->exec_cnt );
+
+    ulong cnt_per_worker   = task_data->info_sz/ctx->exec_cnt;
+    ulong task_infos_gaddr = fd_wksp_gaddr( ctx->runtime_public_wksp, task_data->info );
+
+    for( ulong worker_idx=0UL; worker_idx<ctx->exec_cnt; worker_idx++ ) {
+
+      ulong lt_hash_gaddr = fd_wksp_gaddr( ctx->runtime_public_wksp, &task_data->lthash_values[ worker_idx ] );
+      if( FD_UNLIKELY( !lt_hash_gaddr ) ) {
+        FD_LOG_ERR(( "lt_hash_gaddr is NULL" ));
+        return;
+      }
+
+      ulong start_idx = worker_idx * cnt_per_worker;
+      ulong end_idx   = worker_idx!=ctx->exec_cnt-1UL ? fd_ulong_sat_sub( start_idx + cnt_per_worker, 1UL ) :
+                                                        fd_ulong_sat_sub( task_data->info_sz, 1UL );
+
+      fd_replay_out_ctx_t * exec_out = &ctx->exec_out[ worker_idx ];
+
+      fd_runtime_public_hash_bank_msg_t * hash_msg = (fd_runtime_public_hash_bank_msg_t *)fd_chunk_to_laddr( exec_out->mem, exec_out->chunk );
+      hash_msg->task_infos_gaddr = task_infos_gaddr;
+      hash_msg->lthash_gaddr     = lt_hash_gaddr;
+      hash_msg->start_idx        = start_idx;
+      hash_msg->end_idx          = end_idx;
+
+      fd_stem_publish( stem, exec_out->idx, EXEC_HASH_ACCS_SIG, exec_out->chunk, sizeof(fd_runtime_public_hash_bank_msg_t), 0UL, 0UL, 0UL );
+      exec_out->chunk = fd_dcache_compact_next( exec_out->chunk, sizeof(fd_runtime_public_hash_bank_msg_t), exec_out->chunk0, exec_out->wmark );
+    }
+
+    /* Spins and blocks until all exec tiles are done hashing. */
+    uchar to_check[ ctx->exec_cnt ];
+    uchar to_check_cnt = (uchar)ctx->exec_cnt;
+    for( uchar i=0UL; i<(uchar)ctx->exec_cnt; i++ ) {
+      to_check[ i ] = i;
+    }
+    while( to_check_cnt>0 ) {
+      uchar next_to_check_cnt = 0;
+      for( uchar i=0; i< to_check_cnt; i++ ) {
+        uchar idx = to_check[ i ];
+        ulong res = fd_fseq_query( ctx->exec_fseq[ idx ] );
+        uint state = fd_exec_fseq_get_state( res );
+        if( state != FD_EXEC_STATE_HASH_DONE ) {
+          to_check[ next_to_check_cnt++ ] = idx;
+        }
+      }
+      to_check_cnt = next_to_check_cnt;
+    }
```

Comment:
can simplify by removing the VLA and just have a double-loop

`for (;;) { ... for (...) { if !hash_done continue } break }`